catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -0,0 +1,479 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ExportCommand.class);
+
+    private static final SimpleDateFormat ISO_8601_DATE_FORMAT;
+
+    static {
+        ISO_8601_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH-mm-ss.SSS'Z'");
+        ISO_8601_DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("UTC"));
+    }
+
+    private static final Supplier<String> name =
+            () -> "export-" + ISO_8601_DATE_FORMAT.format(Date.from(Instant.now())) + ".zip";
+
+    private MetacardTransformer transformer;
+
+    private final int PAGE_SIZE = 64;
+
+    private Filter revisionFilter;
+
+    private JarSigner jarSigner = new JarSigner();
+
+    @Reference
+    private StorageProvider storageProvider;
+
+    @Option(name = "--output", description = "Output file to export Metacards and contents into. Paths are absolute and must be in quotes. Will default to auto generated name inside of ddf.home", multiValued = false, required = false, aliases = {
+            "-o"})
+    String output = Paths.get(System.getProperty("ddf.home"), name.get())
+            .toString();
+
+    @Option(name = "--delete", required = true, aliases = {"-d",
+            "delete"}, multiValued = false, description = "Whether or not to delete metacards after export")
+    boolean delete = false;
+
+    @Option(name = "--archived", required = false, aliases = {"-a",
+            "archived"}, multiValued = false, description = "Equivalent to --cql \"\\\"metacard-tags\\\" like 'deleted'\"")
+    boolean archived = false;
+
+    @Override
+    protected Object executeWithSubject() throws Exception {
+        Filter filter = getFilter();
+        transformer = getServiceByFilter(MetacardTransformer.class,
+                String.format("(%s=%s)",
+                        "id",
+                        DEFAULT_TRANSFORMER_ID)).orElseThrow(() -> new CatalogCommandRuntimeException(
+                "Could not get " + DEFAULT_TRANSFORMER_ID + " transformer"));
+        revisionFilter = initRevisionFilter();
+
+        final File outputFile = initOutputFile(output);
+        if (outputFile.exists()) {
+            printErrorMessage(String.format("File [%s] already exists!", outputFile.getPath()));
+            return null;
+        }
+
+        final File parentDirectory = outputFile.getParentFile();
+        if (parentDirectory == null || !parentDirectory.isDirectory()) {
+            printErrorMessage(String.format("Directory [%s] must exist.", output));
+            console.println("If the directory does indeed exist, try putting the path in quotes.");
+            return null;
+        }
+
+        String filename = FilenameUtils.getName(outputFile.getPath());
+        if (StringUtils.isBlank(filename) || !filename.endsWith(".zip")) {
+            console.println("Filename must end with '.zip' and not be blank");
+            return null;
+        }
+
+        SecurityLogger.audit("Called catalog:export command with path : {}\nCurrent user: {}",
```

Comment:
If you need to pass the subject in for some reason (the audit command already pulls the current user), then you should use the correct method to do so: https://github.com/codice/ddf/blob/master/platform/security/platform-security-core-api/src/main/java/ddf/security/common/audit/SecurityLogger.java#L140